### PR TITLE
Indirect Bayes - Post plot and save

### LINKED ISSUE #17020: Indirect Bayes - Post execution plotting and saving
Indirect Interfaces should be able to plot and save after the algorithm is run by clicking a plot and save button rather than having to specify plot and save options before the interface is executed.

This issue should cover the post plotting and saving implementation for the `Indirect > Bayes` interface

* [x] Quasi (blocked by #17012)
* [x] ResNorm
* [x] Stretch

### PR DESCRIPTION
**Pull request by louisemccann**
The plotting and saving of the results workspace has been changed to after the execution of the algorithm.

**To test:**

Files to test with: \olympic\babylon5\Public\Louise McCann\Bayes
For each tab load relevant files and run algorithm. Check each plot option plots a suitable graph and that the appropriate workspaces have been saved in your save directory.
- [x] ResNorm
- [x] Quasi
- [x] Stretch

Fixes #17020.

Release notes will be updated once post plot and save has been completed for all Indirect tabs.

---
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
- [x] Is the code of an acceptable quality?
- [x] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)? Is it well structured with small focussed classes/methods/functions?
- [ ] Are there unit/system tests in place? Are the unit tests small and test the a class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?
##### Functional Tests
- [x] Do changes function as described? Add comments below that describe the tests performed?
- [ ] How do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant documentation been added/updated?
- [ ] Is user-facing documentation written in a user-friendly manner?
- [x] Has developer documentation been updated if required?
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: MantidQt/CustomInterfaces/src/Indirect/Stretch.cpp [hunk 1]
```diff
@@ -219,20 +200,35 @@ void Stretch::saveWorkspaces(const QString &fitWorkspace,
...
+  IndirectTab::checkADSForPlotSaveWorkspace(m_fitWorkspaceName, false);
+
+  WorkspaceGroup_sptr fitWorkspace;
+  fitWorkspace = AnalysisDataService::Instance().retrieveWS<WorkspaceGroup>(
+      m_fitWorkspaceName);
+
+  auto sigma = QString::fromStdString(fitWorkspace->getItem(0)->getName());
+  auto beta = QString::fromStdString(fitWorkspace->getItem(1)->getName());
+  if (sigma.right(5).compare("Sigma") == 0) {
+    if (beta.right(4).compare("Beta") == 0) {
```
Comment (DavidFair) [inline MantidQt/CustomInterfaces/src/Indirect/Stretch.cpp]: This if doesn't do anything is that correct? The `else` block below relates to `sigma.right(5).compare("Sigma") == 0`

**Comment by DavidFair** [comment]
Do the release notes need to be updated to inform users of these new changes?

**Comment by louisemccann** [comment]
Release notes will be updated once post plot and save has been completed for all Indirect tabs.

**Comment by DavidFair** [comment]
Ok I'm happy to :shipit:

**Pull request merged**